### ISSUE TITLE
[CLI] Integrate Wandb SWEEP into Python Script Error

### ISSUE THREAD

**Issue by NguyenAnhTien** [wandb/client#2282]
**Description**
I tried to integrate Wandb sweep into Python script. Inside my script, I call 
wandb.agent(sweep_id, train(), count=5) with the following configuration
sweep_config = {
            'method': 'random',
            'metric' : {
                'name' : 'loss',
                'goal' : 'minimize'
            },
            'parameters' : {
                'optimizer': {
                    'values': ['adam', 'sgd']
                },
                'epochs' : {
                    'values' : [1]
                },
                'lr_rate' : {
                    'values' : [0.01]
                },
                'batch_size' : {
                    'values' : [32, 16]
                },
                'model_name' : {
                    'values' : ['ViT-B_16']
                }
            },
        }
However, for each iteration, Wandb always tried to execute the command line:
 /usr/bin/env python
This command line lead to errors.

**Wandb features**
wandb.agent()
wandb.sweep()

**Environment**
- OS: Redhat 8
- Environment: Redhat Linux Terminal, Jenkin Shell
- Python Version: 3.9

**Comment by vanpelt** [comment]
Hey @NguyenAnhTien can you share the errors / stack trace you're actually seeing?  We generally only recommend using `wandb.agent` inside of jupyter notebooks.  Ideally you would specify the python program to run in your sweep.yaml and then run `wandb agent XXX` from the command line.  

If running from the command line isn't an option, you could try setting the WANDB_START_METHOD=thread.

**Comment by Athe-kunal** [comment]
Hey @vanpelt. I am trying to run PPO from the stable baselines3 and I am encountering this error of  
`
ERROR Detected 5 failed runs in a row at start, killing sweep.
wandb: To change this value set WANDB_AGENT_MAX_INITIAL_FAILURES=val
`
Where can I set the max initial failures values and where do I need to pass it

`project_name = 'finrl-sweeps-sb3'
sweep_config = model_params(model_name)

sweep_id = wandb.sweep(sweep_config,project=project_name)
wandb.agent(sweep_id, train, count=30)`

**Comment by anmolmann** [comment]
@Athe-kunal , could you try setting env var `os.environ["WANDB_AGENT_DISABLE_FLAPPING" = True` in your script?

**Comment by Athe-kunal** [comment]
Hey @anmolmann,  it was expecting a string argument 
`os.environ["WANDB_AGENT_DISABLE_FLAPPING"] = "true"`

I also tried setting 
`os.environ["WANDB_AGENT_MAX_INITIAL_FAILURES"]= "20"`

But still, the PPO failed after the first trial with a value error that the model has diverged. This may be because of sub-optimal hyperparameters which is understandable but even after this setting, there is a problem and it is not going on after first trial.

**Comment by anmolmann** [comment]
@Athe-kunal, could you specify bit more what error you encountered after setting the above env vars. Also, could you please share the debug log files as well? Also, as Chris suggested above we generally only recommend using `wandb.agent` inside of jupyter notebooks. Could you specify the python program to run in your `sweep.yaml` instead. Then, run `wandb sweep sweep.yaml` (this will return you the sweep_id) & `wandb agent sweep_id` from the command line?

**Comment by Athe-kunal** [comment]
@anmolmann, I am trying to implement hyperparameter sweep to a PPO model for FinRL environment https://github.com/AI4Finance-Foundation/FinRL. I have a dictionary of hyperparameters, not a YAML file. So when I run my agent the model diverges and I get a log(zero) error for the policy of my actor, which is expected if I have bad hyperparameters. So I am trying to increase the number of initially failed trials so that the sweep keeps trying after the first failure. But it is stopping after the first failed trial. Also, I am running from jupyter notebooks, not the command line.

**Comment by anmolmann** [comment]
@Athe-kunal , I didn't get any log(error) for the policy. Also, thanks for the update.

**Comment by pkuderov** [comment]
> Also, as Chris suggested above we generally only recommend using `wandb.agent` inside of [jupyter notebooks](https://docs.wandb.ai/guides/sweeps/python-api)

@anmolmann Could you, please, point me where in the docs such division is actually recommended?

I also would like to find discussions on the topic as I don't get such choice and want to see any pros of selecting CLI approach with the sweep ID copy-and-paste shenanigans over clean Python API. So, if you can point me to the right direction, I'd be grateful!

**Comment by Aaronponceuv** [comment]
> @pkuderov , for running sweeps:
> 
> 1. In Jupyter notebooks: [how to(s) are explained in our docs here](https://docs.wandb.ai/guides/sweeps/python-api).
> 2. Via CLI: First, [set up your python script](https://docs.wandb.ai/guides/sweeps/quickstart#set-up-your-python-training-script). Then, [configure your sweep](https://docs.wandb.ai/guides/sweeps/quickstart#2.-configure-your-sweep) and [initialize it](https://docs.wandb.ai/guides/sweeps/quickstart#3.-initialize-a-sweep). Once, this is done, it's time to [launch your agents](https://docs.wandb.ai/guides/sweeps/quickstart#4.-launch-agent-s) via CLI.
>    Also, in our [sweep configuration](https://docs.wandb.ai/guides/sweeps/configuration) guide here, we specifiy that a python data structure is best to run sweeps from jupyter nb.
> 
> > Customize your sweeps. There are two ways to specify the configuration:
> > 
> > 1. [YAML file](https://docs.wandb.com/sweeps/quickstart#2-sweep-config): best for sweeps from the command line. See examples [here](https://github.com/wandb/examples/tree/master/examples/keras/keras-cnn-fashion).
> > 2. [Python data structure](https://docs.wandb.ai/guides/sweeps/python-api): best for running a [sweep from a Jupyter Notebook](https://docs.wandb.ai/guides/sweeps/python-api).
> > 
> > Both methods use the same keys and values, described below.

Hello, for python script jupyter method works ?

**Comment by JakobLS** [comment]
In case it helps anyone; just restart the wandb agent after setting the `WANDB_AGENT_MAX_INITIAL_FAILURES` env variable and it will pick up from the most recent failed run. No need to rerun them all. 

```bash
wandb agent username/project-name/sweep-ID
```
